ciso/_ciso.pyx

```diff
@@ -40,14 +40,15 @@ def zslice(double[:, :, ::1] q,
 
     cdef double[:, ::1] q_iso = np.empty((M, L), dtype=np.float64)
 
-    for i in range(L):
-        for j in range(M):
-            q_iso[j, i] = mask_val
-            for k in range(N-1):
-                if (((p[k, j, i] < p0) and (p[k+1, j, i] > p0)) or
-                   ((p[k, j, i] > p0) and (p[k+1, j, i] < p0))):
-                    dp = p[k+1, j, i] - p[k, j, i]
-                    dp0 = p0 - p[k, j, i]
-                    dq = q[k+1, j, i] - q[k, j, i]
-                    q_iso[j, i] = q[k, j, i] + dq*dp0/dp
+    with nogil:
+        for i in range(L):
+            for j in range(M):
+                q_iso[j, i] = mask_val
+                for k in range(N-1):
+                    if (((p[k, j, i] < p0) and (p[k+1, j, i] > p0)) or
+                       ((p[k, j, i] > p0) and (p[k+1, j, i] < p0))):
+                        q_iso[j, i] = (q[k, j, i] +
+                                       (q[k+1, j, i] - q[k, j, i]) *  # dq
+                                       (p0 - p[k, j, i]) /  # dp0
+                                       (p[k+1, j, i] - p[k, j, i]))  # dp
     return np.array(q_iso)
```

Comment:
I don't like that `np.array` either and I just checked that

```cython
out_arr = np.empty((M, L), dtype=np.float64)
cdef double[:, ::1] q_iso = out_arr
```

works just fine with virtually no performance difference.  So I am changing the code to use that.

Regarding the  memory view. I did read that it should be faster, not to mentioned that we can release the gil.  I haven't tried non-memory view and `prange` though.

Here is a poor's man benchmark:

http://nbviewer.ipython.org/gist/ocefpaf/e69d491fb7e0800dce07